src/test/java/com/google/api/generator/engine/writer/ImportWriterVisitorTest.java

```diff
@@ -54,6 +55,79 @@ public void setUp() {
     writerVisitor = new ImportWriterVisitor(CURRENT_PACKAGE);
   }
 
+  @Test
+  public void writeNewObjectExprImports_basic() {
+    // [Constructing] new ArrayList<>()
+    NewObjectExpr newObjectExpr =
+        NewObjectExpr.builder()
+            .setIsGeneric(true)
+            .setType(TypeNode.withReference(ConcreteReference.withClazz(ArrayList.class)))
+            .build();
+    newObjectExpr.accept(writerVisitor);
+    assertEquals(writerVisitor.write(), "import java.util.ArrayList;\n\n");
+  }
+
+  @Test
+  public void writeNewObjectExprImports_genericsAndVariableArgs() {
+    // [Constructing] new HashMap<List<String>, Integer>>(int initialCapacity, float loadFactor)
```

Comment:
As before, could we please add a test where the argument types need to be imported as well?

Reply:
cool, added a more straightforward real use case from baseline : `new FileOutputStream(File file)` where `FileOutputStream` and `File` both need to be imported. Thanks